cupy_backends/cuda/api/runtime.pyx

```diff
@@ -809,6 +823,21 @@ cpdef streamAddCallback(intptr_t stream, callback, intptr_t arg,
     check_status(status)
 
 
+cpdef launchHostFunc(intptr_t stream, callback, intptr_t arg):
+    if _is_hip_environment:
+        raise RuntimeError('This feature is not supported on HIP')
+    if CUDA_VERSION < 10000:
+        raise RuntimeError('This feature is only supported on CUDA 10.0+')
```

Comment:
@takagi by inspecting the generated C file I noticed Cython (at least 0.29.21 that I'm using) has a nice property that for the `if` conditions that can be determined at compile time, Cython will be smart enough to determine the rest of code is dead and eliminate it! (So for RTD and HIP, this function actually stops right after raising.) As a result, it's fine even if we don't add any stubs (for example I forgot to add stubs for CUDA 9.2 😂) Can we rely on this behavior?

Reply:
Sounds so nice! Would you check if it works with Cython 0.28.0 as, currently, CuPy requires Cython 0.28.0 or later to build it from its source?

Reply:
Hi @takagi, I just checked Cython 0.28.0 will also eliminate the dead code! This is the warning thrown during cythonizing (in both versions): 
```
[50/59] Cythonizing cupy_backends/cuda/api/runtime.pyx
warning: cupy_backends/cuda/api/runtime.pyx:832:16: Unreachable code
```
Though I think 0.28 is too outdated and should be avoided as much as possible (#4148).

Reply:
Can I ask you to post a new issue so other core members can find that?

Reply:
No problem, @takagi, see #4393.

Reply:
Thanks!